run/sigterm-handler/main_test.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"net/http/httptest"
+	"testing"
+)
+
+func TestHandler(t *testing.T) {
+	req := httptest.NewRequest("GET", "/", nil)
+	rr := httptest.NewRecorder()
+	handler(rr, req)
```

Comment:
This test is only checking a HelloWorld handler. Whether handler() returns 'Hello World\n' is not the important thing about this test, and sidesteps the server/signal handling logic in the sample.

Ideally we would:

1. Check the sample can deploy
2. Trigger SIGTERM
3. See SIGTERM is handled

#2 might be done with a "hidden" handler.
#3 might have to be done with log analysis but I can imagine something hacky with global state.

Reply:
Added E2E tests